sdk/core/Azure.Core/tests/TestFramework.props

```diff
@@ -1,38 +0,0 @@
-﻿<Project ToolsVersion="15.0">
-  <PropertyGroup>
-    <SessionRecordsDirectory>SessionRecords</SessionRecordsDirectory>
-    <DefineConstants>$(DefineConstants);TESTFRAMEWORK</DefineConstants>
-  </PropertyGroup>
-
-  <ItemGroup>
-    <Compile Include="$(MSBuildThisFileDirectory)\TestFramework\*.cs" Link="TestFramework\%(RecursiveDir)\%(Filename)%(Extension)" />
-    <Compile Include="$(MSBuildThisFileDirectory)\..\src\Shared\ContentTypeUtilities.cs" Link="TestFramework\ContentTypeUtilities.cs" />
-    <Compile Include="$(MSBuildThisFileDirectory)\..\src\Shared\EventSourceEventFormatting.cs" Link="TestFramework\EventSourceEventFormatting.cs" />
-    <Compile Include="$(MSBuildThisFileDirectory)\..\src\Shared\NullableAttributes.cs" Link="TestFramework\NullableAttributes.cs" Condition="'$(UseAzureCoreNullableAttributes)'!='true'" />
-    <None Update="SessionRecords\**\*.json" CopyToOutputDirectory="PreserveNewest" />
-  </ItemGroup>
-
-  <ItemGroup Condition="'$(ExcludeRecordingFramework)' == 'true'">
-    <Compile Remove="$(MSBuildThisFileDirectory)\TestFramework\LiveOnlyAttribute.cs" />
-    <Compile Remove="$(MSBuildThisFileDirectory)\TestFramework\SamplesBase.cs" />
-    <Compile Remove="$(MSBuildThisFileDirectory)\TestFramework\PlaybackTransport.cs" />
-    <Compile Remove="$(MSBuildThisFileDirectory)\TestFramework\*Record*.cs" />
-    <Compile Remove="$(MSBuildThisFileDirectory)\TestFramework\TestRandom.cs" />
-  </ItemGroup>
-
-  <Target Name="UpdateSessionRecords">
```

Comment:
I think we will still need this target for recording tests. Which means we probably will still need this props file.

Reply:
Let's move the target file into eng folder and conditionally include it for test projects from https://github.com/Azure/azure-sdk-for-net/blob/master/eng/Directory.Build.Data.targets#L105